src/libutil/regex-combinators.hh

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <string_view>
+
+namespace nix::regex {
+
+// TODO use constexpr string building like
+// https://github.com/akrzemi1/static_string/blob/master/include/ak_toolkit/static_string.hpp
+
+static inline std::string either(std::string_view a, std::string_view b)
+{
+    return std::string { a } + "|" + b;
+}
+
+static inline std::string group(std::string_view a)
+{
+    return std::string { "(" } + a + ")";
+}
+
+static inline std::string many(std::string_view a)
+{
+    return std::string { a } + "*";
```

Comment:
This isn't algebraic, as it only works for groups. (Or for the last subexpression, but that can't be right)
Can we add a non-capturing group around `a`?
Same for `either` as well.

Reply:
Ah nice, I forgot about those.

Reply:
```suggestion
    return std::string { "(?:" } + a + ")*";
```